### PR TITLE
FlightTaskOrbit: Improvements

### PR DESCRIPTION
**Pull request by MaEtUgR**
My goal is to take the orbit flight task from a draft to a usable state which can be triggered via SDK **EDIT:** and QGC.

This pr is to fix bugs and improve flight performance.

### CODE REVIEW

File: src/lib/FlightTasks/tasks/FlightTaskOrbit.hpp [hunk 1]
```diff
@@ -41,25 +41,52 @@
...
-	float _z = 0.f; /**< local z coordinate in meters */
-	matrix::Vector2f _center;
+	float _v = 0.f; /**< tangential velocity for orbiting in m/s */
+	matrix::Vector2f _center; /**< local frame coordinates of the center point */
 
+	// TODO: create/use parameters for limits
+	const float _radius_min = 1.f;
+	const float _radius_max = 20.f;
+	const float _velocity_max = 10.f;
+	const float _acceleration_max = 2.f;
```
Comment (Stifael) [inline src/lib/FlightTasks/tasks/FlightTaskOrbit.hpp]: could already replace velocity_max and acceleration_max with the existing parameters. Although, they will have to be adjusted anyway because they will be part of the Flighttask constraints structure

File: src/lib/FlightTasks/tasks/FlightTaskOrbit.cpp [hunk 2]
```diff
@@ -47,51 +47,85 @@ FlightTaskOrbit::FlightTaskOrbit()
...
+
+	// xy velocity to go around in a circle
 	Vector2f center_to_position = Vector2f(_position.data()) - _center;
 	Vector2f velocity_xy = Vector2f(center_to_position(1), -center_to_position(0));
 	velocity_xy = velocity_xy.unit_or_zero();
 	velocity_xy *= _v;
 
-	/* xy velocity adjustment to stay on the radius distance */
+	// xy velocity adjustment to stay on the radius distance
 	velocity_xy += (_r - center_to_position.norm()) * center_to_position.unit_or_zero();
```
Comment (Stifael) [inline src/lib/FlightTasks/tasks/FlightTaskOrbit.cpp]: need to figure out why there is a bug when setting position setpoint

File: src/lib/FlightTasks/tasks/FlightTaskOrbit.cpp [hunk 3]
```diff
@@ -47,51 +47,85 @@ FlightTaskOrbit::FlightTaskOrbit()
...
-	_v -= _sticks_expo(1) * _deltatime;
-	_v = math::constrain(_v, -7.f, 7.f);
-	_z += _sticks_expo(2) * _deltatime;
+	// update altitude
+	FlightTaskManualAltitudeSmooth::update();
 
-	_position_setpoint = Vector3f(NAN, NAN, _z);
+	// stick input adjusts parameters within a fixed time frame
+	const float r = _r + _sticks_expo(0) * _deltatime * (_radius_max / 8.f);
+	const float v = _v - _sticks_expo(1) * _deltatime * (_velocity_max / 4.f);
```
[the remark this answers is not in the document]
Comment (MaEtUgR) [reply src/lib/FlightTasks/tasks/FlightTaskOrbit.cpp]: Explanation:
- full input -> `_sticks_expo` is one
- accumulating `_sticks_expo(0) * _deltatime` increases one every second
- scaling with `(max / reach_time)` results in going from 0 to `max` in `reach_time` seconds

Values should not be hardcoded which is also planned for the introduce parameters step. Is that acceptable? If I remember correctly there are fixes for task parameters in https://github.com/PX4/Firmware/pull/9368 right?

File: src/lib/FlightTasks/tasks/FlightTaskOrbit.cpp (hunk 2 above)
Comment (MaEtUgR) [reply src/lib/FlightTasks/tasks/FlightTaskOrbit.cpp]: Yes, for sure. I just didn't want to break existing smooth functionality. Once the problem is found the change here will be small.

File: src/lib/FlightTasks/tasks/FlightTaskOrbit.hpp (hunk 1 above)
Comment (MaEtUgR) [reply src/lib/FlightTasks/tasks/FlightTaskOrbit.hpp]: That's exactly what I thought... That's the first step.

**Comment by MaEtUgR** [comment]
I'm currently adding the MAVLink interface for Orbit such that it's available with newest QGC daily.
Command: https://github.com/mavlink/mavlink/pull/924
UI: https://github.com/mavlink/qgroundcontrol/pull/6610

**Comment by MaEtUgR** [comment]
In SITL the execution of the MAVLink command now works. Needs still a lot of testing especially also for corner cases like failsafes.
